Classes/commands.py

```diff
@@ -1107,3 +1107,125 @@ async def count_officers(self, ctx):
+        bot = self.bot
+        # Get all fields from LeaveTimes
+        loa_entries = await self.bot.officer_manager.send_db_request('SELECT officer_id, date(date_start), date(date_end), reason, approved FROM LeaveTimes')
+
+        loa_officer_ids = []
+
+        # If the entry is still good, add the officer to our exclusion list. Otherwise, delete the entry if expired.
+        for entry in loa_entries:
+            if entry[2] > datetime.now().date():
+                if entry[3]: loa_officer_ids.append(entry[0])
+            else:
+                await self.bot.officer_manager.send_db_request("DELETE FROM LeaveTimes WHERE officer_id = " + str(entry[0]))
+
+        # For everyone in the server where their role is in the role ladder,
+        # get their last activity times, or if no last activity time, use
+        # the time we started monitoring them. Exclude those we have already
+        # determined have a valid Leave of Absence
+
+        # Get a date range for our LOAs, and make some dictionaries to work in
+        max_inactive_days = bot.settings["max_inactive_days"]
+        oldest_valid = datetime.now() - timedelta(days=max_inactive_days)
+        global inactive_officers
+        inactive_officers = []
+        role_ids = role_id_index(bot.settings)
+        officers_to_check = []
+        guild = self.bot.officer_manager.guild
+
+        for member in guild.members:
+            for role in member.roles:
+                if (
+                    role.id in role_ids
+                    and member.id not in loa_officer_ids
+                    and member not in officers_to_check
+                ):
+                    officers_to_check.append(member)
+                    officer = self.bot.officer_manager.get_officer(member.id)
+                    last_activity = await officer.get_last_activity(
+                        ctx.bot.officer_manager.all_monitored_channels
+                    )
+                    last_activity = last_activity["time"]
+                    # If they've gone past the inactivity limit, add them to the list
+                    if last_activity < oldest_valid:
+                        inactive_officers.append(member)
+
+        # Build the message to send back
+        print(inactive_officers)
+        if len(inactive_officers) == 0:
+            await ctx.channel.send("There are no Officers needing to be marked inactive at this time. It is a good day in the LPD!")
+            return
+        string = "Inactive Officers are"
+        for member in inactive_officers:
+            string = string + " " + member.mention
+        string = (
+            string
+            + ", and may be removed. Run `=mark_inactive` to mark these Officers as `Inactive`."
+        )
+
+        await ctx.channel.send(string)
+
+    @checks.is_admin_bot_channel()
+    @checks.is_white_shirt()
+    @commands.command()
+    # Mark Officers inactive after running =list_inactive
+    async def mark_inactive(self, ctx):
```

Comment:
This command should list out the officers that will be added and ask if it is ok to add them with the Confirm class in menus.py, this is relatively easy to add and you can see how it is done in commands like =link.

Reply:
The list_inactive function has been removed and integrated into the mark_inactive function, with the addition of the menus interface.